torchgeo/datasets/utils.py

```diff
@@ -17,15 +17,23 @@
 from collections.abc import Iterable, Iterator, Sequence
 from dataclasses import dataclass
 from datetime import datetime, timedelta
-from typing import Any, cast, overload
-
+from typing import Any, cast, overload, Union
+from typing_extensions import TypeAlias
 import numpy as np
 import rasterio
 import torch
 from torch import Tensor
 from torchvision.datasets.utils import check_integrity, download_url
 from torchvision.utils import draw_segmentation_masks
 
+Path: TypeAlias = Union[
+    str,
+    bytes,
+    os.PathLike[str],
+    os.PathLike[bytes],
+    Iterable[os.PathLike[str]],
+    Iterable[os.PathLike[bytes]],
```

Comment:
Iterables shouldn't be included, most of the time they aren't supported

Reply:
oh! Thank you sir

Reply:
Sir you have mentioned to add them in this [comment](https://github.com/microsoft/torchgeo/pull/1678#discussion_r1363698094)?